### PR TITLE
Add access control comment for ERC721 _burn function

### LINKED ISSUE #3088: ERC721: document no access control in _burn function
**📝 Details**

The _burn function in ERC721.sol does not have require statement to check if the sender is either owner or approved. If the function is used directly then the user can burn any token. 

current implementation of _burn function: 

    function _burn(uint256 tokenId) internal virtual {
        address owner = ERC721.ownerOf(tokenId);

        _beforeTokenTransfer(owner, address(0), tokenId);

        // Clear approvals
        _approve(address(0), tokenId);

        _balances[owner] -= 1;
        delete _owners[tokenId];

        emit Transfer(owner, address(0), tokenId);

        _afterTokenTransfer(owner, address(0), tokenId);
    }

### PR DESCRIPTION
**Pull request by Niceural**
Fixes #3088  

Adds a doc @dev comment specifying that ERC721 _burn function does not revert if sender is not the owner.

#### PR Checklist

- [x] Tests
- [ ] Documentation
- [ ] Changelog entry

### CODE REVIEW

**Comment by Amxx** [comment]
Hello @Niceural 

This is not part of the documentation because this internal function, and all other internal functions, do not perform any "sender-check". This check is to be included in the public/external method that eventually exposes it as shown [here](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/extensions/ERC721Burnable.sol#L21-L25).

@frangio should we add that?

**Comment by Amxx** [comment]
Ok just saw the related issue.

File: contracts/token/ERC721/ERC721.sol [hunk 1]
```diff
@@ -293,6 +293,7 @@ contract ERC721 is Context, ERC165, IERC721, IERC721Metadata {
     /**
      * @dev Destroys `tokenId`.
      * The approval is cleared when the token is burned.
+     * The current implementation does NOT revert if the sender is not the owner.
```
Comment (Amxx) [inline contracts/token/ERC721/ERC721.sol:296]:
```suggestion
     * This is an internal function that does check if the sender is authorized to operate on the token.
```

Comment (Niceural) [reply contracts/token/ERC721/ERC721.sol:296]: Awesome thanks!

File: contracts/token/ERC721/ERC721.sol [hunk 2]
```diff
@@ -293,6 +293,7 @@ contract ERC721 is Context, ERC165, IERC721, IERC721Metadata {
     /**
      * @dev Destroys `tokenId`.
      * The approval is cleared when the token is burned.
+     * This is an internal function that does check if the sender is authorized to operate on the token.
```
Comment (frangio) [inline contracts/token/ERC721/ERC721.sol:296]: The most important part :stuck_out_tongue: 

```suggestion
     * This is an internal function that does not check if the sender is authorized to operate on the token.
```

**Review by frangio — APPROVED** [review]

**Pull request merged**